cardano-cli/src/Cardano/CLI/EraBased/Run/Governance.hs

```diff
@@ -91,3 +103,146 @@ runGovernanceDRepKeyGen _w vkeyPath skeyPath = firstExceptT GovernanceCmdWriteFi
+
+  eraInMode <- toEraInMode cEra cMode
+    & hoistMaybe (GovernanceCmdEraConsensusModeMismatch (AnyConsensusMode cMode) (AnyCardanoEra cEra))
+
+  constitution <- firstExceptT GovernanceCmdAcqireFailureError
+    ( newExceptT $ executeLocalStateQueryExpr localNodeConnInfo Nothing $ queryConstitution eraInMode sbe)
+      & onLeft (left . GovernanceCmdUnsupportedNtcVersion)
+      & onLeft (left . GovernanceCmdEraMismatch)
+  let serializedGovState = encodePretty constitution
+
+  case mFile of
+    Nothing -> liftIO $ LBS.putStrLn serializedGovState
+    Just (File f) ->
+      handleIOExceptT (WriteFileError . FileIOError f) $
+        LBS.writeFile f serializedGovState
+
+runGovernanceQueryGovState
+  :: ConwayEraOnwards era
+  -> SocketPath
+  -> AnyConsensusModeParams
+  -> NetworkId
+  -> Maybe (File (GovState era) Out)
+  -> ExceptT GovernanceCmdError IO ()
+runGovernanceQueryGovState w socketPath (AnyConsensusModeParams cModeParams) network mFile = conwayEraOnwardsConstraints w $ do
+  let localNodeConnInfo = LocalNodeConnectInfo cModeParams network socketPath
+      sbe = conwayEraOnwardsToShelleyBasedEra w
+      cEra = conwayEraOnwardsToCardanoEra w
+      cMode = consensusModeOnly cModeParams
+
+  eraInMode <- toEraInMode cEra cMode
+    & hoistMaybe (GovernanceCmdEraConsensusModeMismatch (AnyConsensusMode cMode) (AnyCardanoEra cEra))
+
+  govState <- firstExceptT GovernanceCmdAcqireFailureError
+    ( newExceptT $ executeLocalStateQueryExpr localNodeConnInfo Nothing $ queryGovState eraInMode sbe)
+      & onLeft (left . GovernanceCmdUnsupportedNtcVersion)
+      & onLeft (left . GovernanceCmdEraMismatch)
+  let serializedGovState = encodePretty govState
+
+  case mFile of
+    Nothing -> liftIO $ LBS.putStrLn serializedGovState
+    Just (File f) ->
+      handleIOExceptT (WriteFileError . FileIOError f) $
+        LBS.writeFile f serializedGovState
+
+runGetDRepState
+  :: ConwayEraOnwards era
+  -> SocketPath
+  -> AnyConsensusModeParams
+  -> NetworkId
+  -> Set DRepCredential
+  -> Maybe (File () Out)
+  -> ExceptT GovernanceCmdError IO ()
+runGetDRepState w socketPath (AnyConsensusModeParams cModeParams) network drepCreds mFile = conwayEraOnwardsConstraints w $ do
+  let localNodeConnInfo = LocalNodeConnectInfo cModeParams network socketPath
+      sbe = conwayEraOnwardsToShelleyBasedEra w
+      cEra = conwayEraOnwardsToCardanoEra w
+      cMode = consensusModeOnly cModeParams
+
+  eraInMode <- toEraInMode cEra cMode
+    & hoistMaybe (GovernanceCmdEraConsensusModeMismatch (AnyConsensusMode cMode) (AnyCardanoEra cEra))
+
+  drepState <- firstExceptT GovernanceCmdAcqireFailureError
+    ( newExceptT $ executeLocalStateQueryExpr localNodeConnInfo Nothing $ queryDRepState eraInMode sbe drepCreds)
+      & onLeft (left . GovernanceCmdUnsupportedNtcVersion)
+      & onLeft (left . GovernanceCmdEraMismatch)
+  let serializedGovState = undefined {- encodePretty drepState -} -- TODO how to serialize?
```

Comment:
There's not json encoder out of the box here so I guess I'll need to write one by hand?